data/sensors_recent.php

```diff
@@ -166,33 +182,68 @@ function gateway_button($gw_id, $gw_data = null) {
 EOF;
 	}
 
+
 	function output_cell($rowspan, $data) {
 		echo("  <td rowspan=\"$rowspan\"> " . $data . "</td>\n");
 	}
 
+
+	// main loop:
 	while($row = $result->fetch_array(MYSQLI_ASSOC)) {
 		if ($row['message']) {
 			$message = json_decode($row['message'], true);
-			$metadata = $message['metadata'];
 
-			if (!array_key_exists('gateways', $metadata)) {
-				$gateways = [];
+			if ($row['source'] == 'ttn.v2') {
+				$metadata = $message['metadata'];
 
-				// Convert TTN staging to production format
-				foreach ($metadata as $meta) {
-					$meta['gtw_id'] = 'eui-' . strtolower($meta['gateway_eui']);
-					$meta['snr'] = $meta['lsnr'];
-					$gateways[] = $meta;
+				// fix old v2 record-layout:
+				if (!array_key_exists('gateways', $metadata)) {
+					$gateways = [];
+	
+					// Convert TTN staging to production format
+					foreach ($metadata as $meta) {
+						$meta['gtw_id'] = 'eui-' . strtolower($meta['gateway_eui']);
+						$meta['snr'] = $meta['lsnr'];
+						$gateways[] = $meta;
+					}
+	
+					$metadata = [
+						'gateways' => $gateways,
+						'frequency' => $metadata[0]['frequency'],
+						'data_rate' => $metadata[0]['datarate'],
+						'coding_rate' => $metadata[0]['codingrate'],
+					];
```

Comment:
Dit blokje code converteert van v1 naar v2, maar staat nu alleen binnen het source == ttn.v2 blok, dus dat klopt nog niet. Volgens mij moet het iets worden als:

```
if (ttn.v1) {
  // Bovenstaand blok
} else if (ttn.v2) {
  $metadata = $message['metadata'];
} else if (ttn.v3) {
 // Jouw nieuwe code
}
```

Reply:
Of misschien is het nog explicieter om in het geval van ttn.v2 het uit te schrijven:
```
			$metadata = [
					'gateways' => $message['metadata']['gateways'],
					'frequency' => $message['metadata']['frequency'],
					'data_rate' => $message['metadata']['datarate'],
					'coding_rate' => $message['metadata']['codingrate'],
			];
```